packages/lib/reducer.ts

```diff
@@ -1005,7 +1005,11 @@ const reducer = produce((draft: Draft<State> = defaultState, action: any) => {
 
 				draft.notes = newNotes;
 
-				if (noteFolderHasChanged) {
+				// Ensure that the selected note is still in the current folder.
+				// For example, if the user drags the current note to a different folder,
+				// a new note should be selected.
+				// In some cases, however, the selection needs to be preserved (e.g. the mobile app).
+				if (noteFolderHasChanged && !action.preserveSelection) {
 					let newIndex = movedNotePreviousIndex;
```

Comment:
This approach preserves the existing note behavior on desktop (dragging the selected note to a new folder selects a different note) and the [existing back button behavior on mobile](https://github.com/laurent22/joplin/pull/10597#issuecomment-2170330379). However, it makes both the reducer logic and `Note.save` more complicated. 

Suggestions on how this solution might be improved are welcome.

Reply:
Indeed it's not ideal but can't really think of a better way. Let's go with this for now, and at least there's a comment to explain the behaviour and a test to check it.